main.py

```diff
@@ -44,6 +42,7 @@
 binary_bar_preview = Preview((display_position_x, display_position_y), preview_size, "0", game)
 
 score_display = ScoreDisplay("0", (130, display_position_y), 60, game)
+game.score_display = score_display
```

Comment:
Why keep two different references to the same display? One local and one as an attribute of `game`?
I also see later you're passing the local one into functions that already have `game`.